### PR TITLE
MM-50010 - Share common calls code between web and mobile

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- We're extracting rtcpeer, rtcstats, and shared types into their own repo, and importing them into calls plugin and mobile. This will help reduce code duplication.
- Added typings to the WebSocketMessages.

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-50010

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
- Android: 13, Pixel 6
- iOS: 15.7, iPhone 7 plus

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by streamer45 — APPROVED** [review]
This is great as it will simplify a lot of things, thanks for making it happen! :100:

**Review by enahum — COMMENTED** [review]
Other than my comment about the library, it looks good to me.

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "@mattermost/react-native-paste-input": "0.6.2",
     "@mattermost/react-native-turbo-log": "0.2.3",
     "@mattermost/react-native-turbo-mailer": "0.2.4",
+    "@mmcalls/common": "github:mattermost/calls-common#02c52be",
```
[the remark this answers is not in the document]
Comment (cpoile) [reply package.json]: Sure, that's a better name. I went with `@mattermost/calls`. I tagged the version as well. I'm not sure about publishing to npm though, what would the benefit of that be?

[the remark this answers is not in the document]
Comment (enahum) [reply package.json]: I guess is easier to maintain and keep each project isolated and in sync without having to reference a specific commit, makes it cleaner in my opinion, the effort should really be minimal.

but we can keep it like this for now, have a convo with your team and decide if it makes sense to publish it.

**Review by enahum — APPROVED** [review]

**Pull request merged**